packages/backend-core/src/sql/sql.ts

```diff
@@ -602,24 +605,33 @@ class InternalBuilder {
     } else if (paginate && paginate.limit) {
       foundLimit = paginate.limit
     }
-    // start building the query
-    let query = this.knexWithAlias(knex, endpoint, tableAliases)
-    query = query.limit(foundLimit)
-    if (foundOffset) {
+    // always add the found limit, unless counting
+    if (!counting) {
+      query = query.limit(foundLimit)
+    }
```

Comment:
The large number of `if (!counting)` conditions in this function makes me feel like this should actually be 2 separate functions. What do you think?